File: python/sprague_grundy.py
# BOJ 13034 다각형 게임

import sys
input = sys.stdin.readline

n = int(input())

grundy = [0] * (n + 1)
for i in range(2, n + 1):
    v = [0] * (i + 1)
    left = 0
    right = i - 2
    while left <= right:
        v[grundy[left] ^ grundy[right]] = True
        left += 1
        right -= 1
    
    mex = 0
    while v[mex]:
        mex += 1
    
    grundy[i] = mex

print(2 if grundy[n] == 0 else 1)

File: sprague_grundy.cpp
// BOJ 13034 다각형 게임

#include <bits/stdc++.h>
#define sz size()
#define bk back()
#define fi first
#define se second

using namespace std;
typedef long long ll;
typedef pair<int, int> pii;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);

    int n;
    cin >> n;

    vector<int> grundy(n + 1);
    for (int i = 2; i <= n; i++) {
        vector<bool> v(i + 1);
        int left = 0;
        int right = i - 2;
        while (left <= right) {
            v[grundy[left] ^ grundy[right]] = true;
            left++;
            right--;
        }

        int mex = 0;
        while (v[mex])
            mex++;

        grundy[i] = mex;
    }

    cout << (grundy[n] == 0 ? 2 : 1);
}
